test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -86,887 +86,962 @@ define([
                 });
             });
 
-            // message with a mix of valid and invalid data
-            it(`should separate out valid and invalid ${type} data`, () => {
-                const msg = makeCommMsg(msgType, validAndInvalidData),
-                    comm = new JobCommChannel();
+            it('should create a job mapping from job info data', () => {
+                const infoData = ResponseData[jcm.MESSAGE_TYPE.INFO];
+                const comm = new JobCommChannel(),
+                    msg = makeCommMsg(jcm.MESSAGE_TYPE.INFO, infoData);
+
                 return comm.initCommChannel().then(() => {
-                    spyOn(comm, 'reportCommMessageError');
-                    spyOn(console, 'error');
-                    spyOn(testBus, 'send');
                     comm.handleCommMessages(msg);
-                    expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
-                        [{ msgType, msgData: expected.invalid }],
-                    ]);
+                    expect(comm._jobMapping).toEqual(responseDataMapping);
+                });
+            });
 
-                    const expectedMessages = Object.keys(expected.valid).map((key) => {
-                        return [
-                            expected.valid[key],
-                            {
-                                channel: { [JOB_CHANNEL]: key },
-                                key: { type: msgType },
-                            },
-                        ];
-                    });
+            // since the retry data includes the job status,
+            // we can reconstitute the job mapping from it
+            it('should create a job mapping from job retry data', () => {
+                const retryData = ResponseData[jcm.MESSAGE_TYPE.RETRY];
+                const comm = new JobCommChannel(),
+                    msg = makeCommMsg(jcm.MESSAGE_TYPE.RETRY, retryData);
 
-                    expect(testBus.send.calls.allArgs()).toEqual(
-                        jasmine.arrayWithExactContents(expectedMessages)
-                    );
+                return comm.initCommChannel().then(() => {
+                    comm.handleCommMessages(msg);
+                    expect(comm._jobMapping).toEqual(responseDataMapping);
+                });
+            });
+
+            it('should create and maintain a job mapping', () => {
+                const batchUpdates = JobsData.batchJob.jobUpdateSeries;
+                const comm = new JobCommChannel();
+                return comm.initCommChannel().then(() => {
+                    spyOn(testBus, 'send');
+                    batchUpdates.forEach((update) => {
+                        comm.handleCommMessages(makeCommMsg(update.type, update.msg));
+                        expect(Object.keys(comm._jobMapping)).toEqual(
+                            jasmine.arrayWithExactContents(update.allJobIds)
+                        );
```

Comment:
all that data preparation and the test itself seems so asinine! how disappointing.

Reply:
Not asinine, helpful!